activerecord/lib/active_record/relation/predicate_builder/polymorphic_array_value.rb

```diff
@@ -42,10 +42,16 @@ def klass(value)
         end
 
         def convert_to_id(value)
+          primary_key = primary_key(value)
+
           if value.is_a?(Base)
```

Comment:
Maybe it's better to move the primary key assignment inside the if clause so that it's not called when hitting the else clause.

Reply:
Yeah, it was for DRYing the code, but turned out this is not working, thanks.